Formula/libxml2.rb

```diff
@@ -41,7 +41,7 @@ class Libxml2 < Formula
 
   keg_only :provided_by_osx
 
-  depends_on :python => :optional
+  depends_on :python if MacOS.version <= :snow_leopard
```

Comment:
@tdsmith I don't think it's unrelated to this PR. If we're going to have some more formulae with bindings to the system Python and some to Homebrew's Python we're going to end up with even more confused users.

> I favor my homebrew-evolution proposal.

If we were coming to it today I'd be 👎 on that being merged. Since it was merged I've had multiple users complain to me about how installing random things and having `which python` change has caused confusion and wasted their time. I'm not sure how much longer we can justify Python being a special snowflake when it comes to both `@` versioning, system duplication and the keg-only status that comes alongside that.